src/internal/collection/postgres_collection.go

```diff
@@ -0,0 +1,711 @@
+package collection
+
+import (
+	"context"
+	"crypto/md5"
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	etcd "github.com/coreos/etcd/clientv3"
+	"github.com/gogo/protobuf/proto"
+	"github.com/jackc/pgerrcode"
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/watch"
+	"github.com/pachyderm/pachyderm/v2/src/version"
+)
+
+const (
+	modifiedTriggerName  = "update_modified_trigger"
+	modifiedFunctionName = "update_modified_time"
+	watchTriggerName     = "notify_watch_trigger"
+	watchBaseName        = "pwc" // "Pachyderm Watch Channel"
+	indexBaseName        = "idx"
+	pgIdentBase64Values  = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_$"
```

Comment:
That dollar sign seems like a problem.  We can't use `-`?  Postgres uses $ for passing arguments.

Reply:
What about base32?  128bits/16bytes of entropy takes up ~25 bytes. That's 64bit collision resistance.  That should be enough right?  With the twc_ it's still less than the weird 31 character issue I was running into.

Reply:
Oh dammit - this was from and old implementation anyway and I need to remove it - I started using hex-encoded md5 sums instead.